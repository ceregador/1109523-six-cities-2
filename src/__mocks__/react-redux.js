export const connectAdvanced = jest.fn().mockImplementation((component) => component);
const mockConnect = () => connectAdvanced;

export const connect = jest.fn().mockImplementation(mockConnect);
