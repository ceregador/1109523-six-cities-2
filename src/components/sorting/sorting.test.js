import React from 'react';
import {connect, connectAdvanced} from 'react-redux';
import Sorting from '../sorting/sorting.jsx';
import Renderer from 'react-test-renderer';

jest.mock(`react-redux`);

it(`renders and connects correctly`, () => {
  const tree = Renderer
    .create(<Sorting onChangeSortingType={() => null}/>)
    .toJSON();

  expect(connect).toHaveBeenCalledTimes(1);
  expect(connect).toHaveBeenCalledWith(null, expect.any(Object));

  const mapDispatchToProps = connect.mock.calls[0][1];
  expect(mapDispatchToProps).toHaveProperty(`onChangeSortingType`);

  expect(connectAdvanced).toHaveBeenCalledWith(Sorting);

  expect(tree).toMatchSnapshot();
});
