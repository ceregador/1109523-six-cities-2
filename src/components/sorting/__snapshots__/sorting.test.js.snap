// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders and connects correctly 1`] = `
<form
  action="#"
  className="places__sorting"
  method="get"
>
  <span
    className="places__sorting-caption"
  >
    Sort by 
  </span>
  <select
    className="places__sorting-type"
    id="places-sorting"
    onChange={[Function]}
  >
    <option
      className="places__option"
      value="Popular"
    >
      Popular
    </option>
    <option
      className="places__option"
      value="Price: low to high"
    >
      Price: low to high
    </option>
    <option
      className="places__option"
      value="Price: high to low"
    >
      Price: high to low
    </option>
    <option
      className="places__option"
      value="Top rated first"
    >
      Top rated first
    </option>
  </select>
</form>
`;
